rootfs/scheduler/__init__.py

```diff
@@ -1420,5 +1506,292 @@ def get_node(self, name, **kwargs):
 
         return response
 
+    # DEPLOYMENTS #
+
+    def get_deployment(self, namespace, name):
+        url = self._api("/namespaces/{}/deployments/{}", namespace, name)
+        response = self.session.get(url)
+        if unhealthy(response.status_code):
+            raise KubeHTTPException(
+                response,
+                'get Deployment "{}" in Namespace "{}"', name, namespace
+            )
+
+        return response
+
+    def get_deployments(self, namespace, **kwargs):
+        url = self._api("/namespaces/{}/deployments", namespace)
+        response = self.session.get(url, params=self._selectors(**kwargs))
+        if unhealthy(response.status_code):
+            raise KubeHTTPException(response, 'get Deployments in Namespace "{}"', namespace)
+
+        return response
+
+    def _wait_until_deployment_is_updated(self, namespace, name):
+        """
+        Looks at status/observedGeneration and metadata/generation and
+        waits for observedGeneration >= generation to happen
+
+        http://kubernetes.io/docs/user-guide/deployments/#the-status-of-a-deployment
+        More information is also available at:
+        https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#metadata
+        """
+        logger.debug("waiting for Deployment {} to get a newer generation (30s timeout)".format(name))  # noqa
+        for _ in range(30):
```

Comment:
Not inherently, I never figure out a good way to express it was more than 30 seconds but a variable of how long each `get_deployment` would take. I guess it's less of a timeout and more of a 30 attempts